src/Tizen.Sensor/Tizen.Sensor/SensorRecorder.cs

```diff
@@ -0,0 +1,504 @@
+        /// Value.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void OptionSetInt(int recorderOption, int value)
+        {
+            if (CheckOption())
+            {
+                int error = Interop.SensorRecoder.OptionSetInt(OptionHandle, recorderOption, value);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error checking sensor recorder option set int");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.OptionSetInt Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder option is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Sets an integer-type query parameter.
+        /// </summary>
+        /// <param name="recorderQuery">
+        /// Query parameter.
+        /// </param>
+        /// <param name="value">
+        /// Value.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void QuerySetInt(int recorderQuery, int value)
+        {
+            if (CheckQuery())
+            {
+                int error = Interop.SensorRecoder.QuerySetInt(QueryHandle, recorderQuery, value);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error checking sensor recorder query set int");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.QuerySetInt Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Sets a time-type query parameter.
+        /// </summary>
+        /// <param name="recorderQuery">
+        /// Query parameter.
+        /// </param>
+        /// <param name="time">
+        /// Time.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void QuerySetTime(int recorderQuery, int time)
+        {
+            if (CheckQuery())
+            {
+                int error = Interop.SensorRecoder.QuerySetTime(QueryHandle, recorderQuery, time);
+                if (error != (int)SensorError.None)
+                {
+                    Log.Error(Globals.LogTag, "Error checking sensor recorder query set time");
+                    throw SensorErrorFactory.CheckAndThrowException(error, "SensorRecorder.QuerySetTime Failed");
+                }
+            }
+            else
+            {
+                Log.Error(Globals.LogTag, "Recorder query is not created");
+            }
+        }
+
+        /// <summary>
+        ///  Queries the recorded data synchronously.
+        /// </summary>
+        /// <param name="type">
+        /// Sensor type.
+        /// </param>
+        /// <since_tizen> 6 </since_tizen>
+        public void Read(int type)
```

Comment:
Why no result? Usually read operation expect a result of read

Reply:
I am using Event to dispatch the result.